cmd/hypper/lint.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright The Helm Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"github.com/Masterminds/log-go"
+	logio "github.com/Masterminds/log-go/io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/rancher-sandbox/hypper/pkg/action"
+	"helm.sh/helm/v3/pkg/cli/values"
+	"helm.sh/helm/v3/pkg/getter"
+)
+
+var longLintHelp = `
+This command takes a path to a chart and runs a series of tests to verify that
+the chart is well-formed.
+
+If the linter encounters things that will cause the chart to fail installation,
+it will emit [ERROR] messages. If it encounters issues that break with convention
+or recommendation, it will emit [WARNING] messages.
+`
+
+func newLintCmd(logger log.Logger) *cobra.Command {
+	client := action.NewLint()
+	valueOpts := &values.Options{}
+	wInfo := logio.NewWriter(logger, log.InfoLevel)
+
+	cmd := &cobra.Command{
+		Use:   "lint PATH",
+		Short: "examine a chart for possible issues",
+		Long:  longLintHelp,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			paths := []string{"."}
+			if len(args) > 0 {
+				paths = args
+			}
+			if client.WithSubcharts {
+				for _, p := range paths {
+					_ = filepath.Walk(filepath.Join(p, "charts"), func(path string, info os.FileInfo, err error) error {
+						if info != nil {
+							if info.Name() == "Chart.yaml" {
+								paths = append(paths, filepath.Dir(path))
+							} else if strings.HasSuffix(path, ".tgz") || strings.HasSuffix(path, ".tar.gz") {
+								paths = append(paths, path)
+							}
+						}
+						return nil
+					})
+				}
+			}
+
+			client.Namespace = settings.Namespace()
```

Comment:
open question, is this enough? do we need to change the `client.Config` too?

Reply:
for linting yes, as it doesn't really do anything with the client namespace, everything is local. Probably would be able to even remove it...but its left like that because its the same upstream

Reply:
true! thanks for the explanation :)